internal/services/environment/resource_environment.go

```diff
@@ -146,6 +151,20 @@ func (r *Resource) Schema(ctx context.Context, req resource.SchemaRequest, resp
 				Required:            true,
 				Validators: []validator.String{
 					stringvalidator.OneOf(EnvironmentTypes...),
+					validators.OtherFieldRequiredWhenValueOf(path.Root("owner_id").Expression(), nil, regexp.MustCompile(EnvironmentTypesDeveloperOnlyRegex), "owner_id must be set when environment_type is `Developer`"),
```

Comment:
same as other comment, feel like you could do this with a combination of Any / All validators